nio_command_block.py

```diff
@@ -0,0 +1,173 @@
+from nio.metadata.properties import IntProperty, ListProperty, \
+    SelectProperty, ObjectProperty, PropertyHolder, StringProperty, \
+    TimeDeltaProperty, ExpressionProperty
+from nio.common.block.base import Block
+from nio.common.discovery import Discoverable, DiscoverableType
+from nio.common.signal.base import Signal
+from nio.modules.scheduler import Job
+from urllib.parse import urlencode
+import base64
+from .oauth2_mixin.oauth2 import OAuth2, OAuth2Exception
+import requests
+from enum import Enum
+
+
+class SecurityMethod(Enum):
+    NONE = 'none'
+    BASIC = 'basic'
+    OAUTH = 'oauth'
+
+
+class URLParameter(PropertyHolder):
+    prop_name = ExpressionProperty(title="Property Name")
+    prop_value = ExpressionProperty(title="Property Value")
+
+
+class BasicAuthCreds(PropertyHolder):
+    username = StringProperty(title='Username')
+    password = StringProperty(title='Password')
+
+
+@Discoverable(DiscoverableType.block)
+class NioCommand(Block, OAuth2):
+
+    params = ListProperty(URLParameter,
+                          title="Command Parameters",
+                          default=[])
+    host = StringProperty(title="n.io Host", default="[[NIOHOST]]")
+    port = IntProperty(title="n.io Port", default="[[NIOPORT]]")
+    service_name = ExpressionProperty(title="Service Name")
+    block_name = ExpressionProperty(title="Block Name (optional)")
+    command_name = ExpressionProperty(title="Command Name")
+    security_method = SelectProperty(SecurityMethod,
+                                     default=SecurityMethod.BASIC,
+                                     title='Security Method')
+    basic_auth_creds = ObjectProperty(BasicAuthCreds,
+                                      title='Credentials (BasicAuth)')
+
+    # We should periodically re-authenticate with Google, this is the interval
+    # to do so.
+    # Ideally, we use the expiry time in the OAuth token that we get back, but
+    # that will require a non-backwards compatible change to the OAuth2 mixin,
+    # so for now, having an extra non-visible property will have to do
+    reauth_interval = TimeDeltaProperty(
+        title="Reauthenticate Interval",
+        visible=False,
+        default={'seconds': 2400})  # Default to 40 mins
+
+    def __init__(self):
+        super().__init__()
+        self._access_token = None
+        self._reauth_job = None
+
+    def configure(self, context):
+        super().configure(context)
+        if self.security_method == SecurityMethod.OAUTH:
+            self._init_access_token()
+
+    def process_signals(self, signals):
+        for signal in signals:
+            try:
+                url, headers = self._get_url(signal)
+                resp = requests.get(url, headers=headers)
+                sigs = self._process_response(resp)
+                if sigs:
+                    self.notify_signals(sigs)
+            except Exception as e:
+                self._logger.exception(e)
```

Comment:
We don't actually need to pass the exception to logger.exception. Instead, you give it an error message (something like "Could not complete request") and it will automatically add the exception information. For the same reason, unless you are doing something else with `e`, just putting `except:` is sufficient in this case.